torch/csrc/jit/interpreter.cpp

```diff
@@ -692,16 +692,19 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
           });
 
           return true;
-        } catch(std::exception & e) {
-          if (!instructions[pc].debug_location) {
-            throw;
-          }
-          auto msg = instructions[pc].debug_location->wrapException(e, "operation failed in interpreter");
-          if (dynamic_cast<JITException *>(&e)) {
-            throw JITException(msg);
-          } else {
-            throw std::runtime_error(msg);
+        } catch (Future::FutureError& e) {
+          // Error from the forked thread
+          handleError(e.error_msg);
+          return false;
+        } catch (std::exception& e) {
+          // Error from the current thread
+          std::string msg = "";
+          if (instructions[pc].debug_location) {
+            msg = instructions[pc].debug_location->wrapException(
+                e, "operation failed in interpreter");
           }
+          handleError(msg);
```

Comment:
This line effectively deletes the `what()` from the original exception. We need to preserve that.

Reply:
Assume this is what you mean?
```c++
if (instructions[pc].debug_location) {
  handleError(instructions[pc].debug_location->wrapException(
    e, "operation failed in interpreter"));
} else {
  handleError(e.what());
}